cmake/modules/RootConfiguration.cmake

```diff
@@ -800,7 +806,7 @@ else()
     ${CMAKE_BINARY_DIR}/ginclude/compiledata.h "${CMAKE_CXX_COMPILER}"
         "${CMAKE_CXX_FLAGS_RELEASE}" "${CMAKE_CXX_FLAGS_DEBUG}" "${CMAKE_CXX_FLAGS}"
         "${CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS}" "${CMAKE_EXE_FLAGS}" "so"
-        "${libdir}" "-lCore" "-lRint" "${incdir}" "" "" "${ROOT_ARCHITECTURE}" "")
+        "${libdir}" "-lCore" "-lRint" "${incdir}" "" "" "${ROOT_ARCHITECTURE}" "${ROOTBUILD}")
```

Comment:
Alternatively we could pass CMAKE_BUILD_TYPE directly and update https://github.com/root-project/root/blob/f69c9e7ca56411863c1f14ac0334911ec3e33668/core/base/src/TSystem.cxx#L2901 accordingly.

Reply:
@pcanal do you think we can merge this PR like this or you would prefer alternative solution? Honestly, I would prefer a less invasive solution.

Reply:
What do you mean by less invasive? (I.e. I am not sure which solution you prefer :) )

Reply:
@pcanal I would prefer to use only CMake for this